### DISCUSSION TITLE
Radial displacement asymmetry in cylinder thermomechanical coupling when using displaced mesh

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#21858] [Q&A Modules: Solid mechanics] [answered]
Hi, MOOSE experts.

In thermomechanical coupling, the size of the mesh changes when the object is thermally expanded, and the change in the mesh also affects the temperature field. When I set `use_displaced_mesh=true` in all blocks, the distribution of `disp_x` is not symmetric about the x-axis. Likewise, the distribution of `disp_y` is not symmetric about the y-axis. 
![x](https://user-images.githubusercontent.com/99813361/184898117-c19055f5-7d02-40bb-813f-5aea1813a478.png)
![y](https://user-images.githubusercontent.com/99813361/184898158-cf094fcc-c765-4b20-80b5-06005eb9f778.png)

Then I set `use_displaced_mesh=false`, the rest of the input file is not modified. The output disp_x and disp_y are symmetric about the x and y axes. 
![x](https://user-images.githubusercontent.com/99813361/184898983-e2c0db34-b723-47da-a5ec-64c26447d31d.png)
![y](https://user-images.githubusercontent.com/99813361/184899010-8360beb0-cdb0-4b21-81b1-974462d9325a.png)

Why are disp_x and disp_y not symmetrical when `use_displaced_mesh=true` is set? And I've found that the default value of `use_displaced_mesh` varies in different objects. For example, the default value of `use_displaced_mesh` for `ADHeatConductionTimeDerivative` is true, while that for `ADCoupledForce` is false. Should I set `use_displaced_mesh=true` for all objects if I want to simulate thermal coupling as accurately as possible?

All input and output files are attached.
[use_displaced_mesh_true.zip](https://github.com/idaholab/moose/files/9351845/use_displaced_mesh_true.zip)
[use_displaced_mesh_false.zip](https://github.com/idaholab/moose/files/9351846/use_displaced_mesh_false.zip)

**Comment by GiudGiud** [comment, marked as answer]
@recuero

**Reply by GiudGiud** [reply]
I think this was answered somewhere else. Be mindful of solid body modes/rotations was the conclusion

**Reply by js-jixu** [reply]
Yeah, @hugary1995 has solved this problem.https://github.com/idaholab/moose/discussions/21878#discussion-4314967
